src/pystack/_pystack/elf_common.cpp

```diff
@@ -144,13 +118,128 @@ CoreFileAnalyzer::~CoreFileAnalyzer()
     close(d_fd);
 }
 
+void
+CoreFileAnalyzer::removeModuleIf(std::function<bool(Dwfl_Module*)> predicate)
+{
+    using Predicate = decltype(predicate);
+    struct CallbackArgs
+    {
+        Dwfl* dwfl;
+        Predicate& predicate;
+    } callback_args = {d_dwfl.get(), predicate};
+
+    // Remove all modules, except for any that the callback re-adds.
+    dwfl_report_begin(d_dwfl.get());
+
+    int rc = dwfl_report_end(
```

Comment:
Nit (feel free to ignore)

```suggestion
    const int rc = dwfl_report_end(
```